Sources/Plasma/FeatureLib/pfConsole/pfConsole.cpp

```diff
@@ -693,21 +626,39 @@ void    pfConsole::IHandleKey( plKeyEventMsg *msg )
             }
         }
         // or are they just typing in a working line
-        else if( strlen( fWorkingLine ) < kMaxCharsWide - 2 && key != 0 )
+        else
         {
-            for( i = strlen( fWorkingLine ) + 1; i > fWorkingCursor; i-- )
-                fWorkingLine[ i ] = fWorkingLine[ i - 1 ];
-
-            // TODO: What about keys outside of `char` range?
-            fWorkingLine[fWorkingCursor++] = char(key);
-
+            // FIXME: make the console unicode friendly.
+            IHandleCharacter((char)key);
             findAgain = false;
             findCounter = 0;
             IUpdateTooltip();
```

Comment:
It should be fine - the difference will be that any time someone types in a null character (can you actually do this?) or the line is too long, the tooltip will get regenerated. I can guard against that happening to prevent odd semantics, however.